### ISSUE TITLE
[Bug]: CenterNorm() not expanding bounds for negative values

### ISSUE THREAD

**Issue by SethBuesing** [matplotlib/matplotlib#27223]
### Bug summary

When applying CenterNorm, the bounds will only expand to encompass _positive_ value furthest from 0 when the function to map colors is not explicitly specified. In simple plots like scatter this can be circumvented using the `c` keyword argument but for more complicated programs without a good analogue to c it is not clear how to correct this (for example quiver has an optional C argument but this does _not_ adjust the CenterNorm), though sometimes autoscaling the CenterNorm after the plot has been called has worked.

### Code for reproduction

```python
import matplotlib.colors as colors
import matplotlib.pyplot as plt
norm = colors.CenteredNorm()
plt.scatter([0,1,3],[-1,-2,1],norm = norm)
plt.colorbar()
```

### Actual outcome

![image](https://github.com/matplotlib/matplotlib/assets/112661629/c6ab55cd-0d7d-4dfb-9352-42707e1aeb92)

### Expected outcome

![image](https://github.com/matplotlib/matplotlib/assets/112661629/51520793-e072-45b0-802d-a362b9786c6e)

### Additional information

I noticed this bug while using a class function that created a quiver plot on a prescribed grid, with the arrows' colors representing values at that point. I allowed the class function to pass a norm to the quiver, but even though the color argument `C` was filled in the quiver the norm was not expanding the cmap bounds completely if the most distant value from 0 was negative. The minimal example here can be fixed by using 
`plt.scatter([0,1,3],[-1,-2,1],c=[-1,-2,1],norm = norm)`
but the `c` keyword does not have an analogue in quiver plots, other than `[C]` which didn't do the trick.

### Operating system

Windows 10 Professional

### Matplotlib Version

3.7.1

### Matplotlib Backend

module://matplotlib_inline.backend_inline

### Python version

3.10.5

### Jupyter version

3.4.3

### Installation

conda

**Comment by greglucas** [comment]
In your first example with scatter you aren't specifying any _values_ to colormap, you are only specifying the points to put the markers at. You have found out already that you have to explicitly pass your values in with the `c` keyword argument which you put later,`plt.scatter([0,1,3],[-1,-2,1],c=[-1,-2,1],norm = norm)`, so I'm not sure if there is another issue you're identifying.

For the quiver case, it is a different API and you need to pass the U, V, C parameters as arguments to the function, which works to expand the centered norm to (-2, 2) as expected.

```python
u, v, c = [0, 1, 3], [-1, -2, 1], [-1, -2, 1]
plt.quiver(u, v, c, norm=norm)
```

Here is an example coloring the quiver arrows for reference as well: https://matplotlib.org/stable/gallery/images_contours_and_fields/quiver_demo.html#sphx-glr-gallery-images-contours-and-fields-quiver-demo-py

**Comment by SethBuesing** [comment]
I see that the example I provided was not representative of the issue I was observing in my codebase, but I was hesitant to share the entire example. I was able to sort the issue out myself: I was using a subroutine that called quiver and I had forgotten to remove the clim arguments that were passed from the start, which clashed with the CenteredNorm(). I apologize for the unnecessary opening of an issue.
